step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesControllerPlugin.java

```diff
@@ -95,12 +119,101 @@ public void serverStart(GlobalContext context) {
 		GaugeCollectorRegistry.getInstance().registerHandler(handler);
 
 		WebApplicationConfigurationManager configurationManager = context.require(WebApplicationConfigurationManager.class);
-		configurationManager.registerHook(s -> Map.of(RESOLUTION_PERIOD_PROPERTY, resolutionPeriod.toString()));
+		configurationManager.registerHook(s -> Map.of(RESOLUTION_PERIOD_PROPERTY, String.valueOf(timeSeries.getDefaultCollection().getResolution())));
+	}
+
+	private List<TimeSeriesCollection> prepareTimeSeriesCollections(TimeSeriesCollectionsSettings collectionsSettings, CollectionFactory collectionFactory) {
+		HashMap<TimeSeriesCollection, Boolean> collectionsEnabled = new HashMap<>();
+
+		collectionsEnabled.put(
+				new TimeSeriesCollection(collectionFactory.getCollection(TIME_SERIES_MAIN_COLLECTION, Bucket.class),
+						new TimeSeriesCollectionSettings()
+								.setResolution(collectionsSettings.getMainCollectionResolution())
+								.setIngestionFlushingPeriodMs(collectionsSettings.getMainCollectionFlushPeriod())
+								.setMergeBucketsOnIngestionFlush(true)
```

Comment:
Is the merge required for the main collection, can we avoid this?

Reply:
yes, definitely